addons/html_editor/static/src/editor/toolbar/toolbar_plugin.js

```diff
@@ -44,10 +44,10 @@ export class ToolbarPlugin extends Plugin {
     }
 
     handleSelectionChange() {
-        const sel = window.getSelection();
-        const range = sel.rangeCount ? sel.getRangeAt(0) : false;
+        const sel = this.shared.getEditableSelection();
+        const range = sel && sel.rangeCount ? sel.getRangeAt(0) : false;
```

Comment:
`getEditableSelection` returns `undefined` if `rangeCount` is 0.  (This should be probably be written in its JSDoc?)
So the `sel && sel.rangeCount` check is redundant.

Reply:
yes, but if `sel === undefined`, then sel.rangeCount crashes?

Reply:
what I meant is that you don't need the **second** check.
`const range = sel ? sel.getRangeAt(0) : false`
or
`const range = sel?.getRangeAt(0)`
or even
`this.updateToolbarVisibility(sel?.getRangeAt(0));`